### PR TITLE
Allow rural farm spawns

### PR DESCRIPTION
**Pull request by FarkasKBence**
#### Summary

None

#### Purpose of change

With the changes to world generation, and the addition of rural areas, it has become clear that farms don't got much of a range in their city distances scale.

#### Describe the solution

Increased farms' and lumbermills' maximum allowed distance from towns.

Split into two categories, orchards, horse farms were given a maximum of 120, while dairy farms, farm mutables, and lumbermills were given a maximum of 400 tiles; a tenfold increase for some locations.  (arbitrary numbers!)

#### Describe alternatives you've considered

Adjusting the scales even more.  Setting the scale to -1.  Allowing nether monster corpses to spawn in rural areas.

#### Testing

Scrolled south-west in a new world.

#### Additional context

I don't believe in commercial locations having a -1 in their city distances scale.

### CODE REVIEW

**Comment by KeremBabaG** [comment]
400 is too much imo, how about 200?

File: data/json/overmap/overmap_mutable/farm_horse_mutable.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
     "id": "Horse Farm Mutable",
     "subtype": "mutable",
     "locations": [ "forest_without_trail", "field", "open_air" ],
-    "city_distance": [ 5, 40 ],
+    "city_distance": [ 5, 120 ],
```
Comment (Maleclypse) [inline data/json/overmap/overmap_mutable/farm_horse_mutable.json:7]:
```suggestion
    "city_distance": [ 5, 120 ],
```

Does -1 work for infinite distance in city distance?  If it does we should use that for things that should be anywhere  in rural regions.

Comment (Procyonae) [reply data/json/overmap/overmap_mutable/farm_horse_mutable.json:7]: Yeah -1 works fine

**Comment by Maleclypse** [comment]
> 400 is too much imo, how about 200?

I think you are confused.  Why would you think that?

File: data/json/overmap/overmap_mutable/farm_horse_mutable.json (hunk 1 above)
Comment (LovamKicsiGazsii) [reply data/json/overmap/overmap_mutable/farm_horse_mutable.json:7]: On further thought, I agree on these having an infinite range.  For consistency sake.

Comment (Maleclypse) [inline data/json/overmap/overmap_mutable/farm_horse_mutable.json:7]:
```suggestion
    "city_distance": [ 5, -1 ],
```

File: data/json/overmap/overmap_mutable/farm_mutable.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
     "id": "Farm Mutable",
     "subtype": "mutable",
     "locations": [ "subterranean_empty", "land", "open_air" ],
-    "city_distance": [ 5, 40 ],
+    "city_distance": [ 5, 400 ],
```
Comment (Maleclypse) [inline data/json/overmap/overmap_mutable/farm_mutable.json:7]:
```suggestion
    "city_distance": [ 5, -1 ],
```

File: data/json/overmap/overmap_special/specials.json [hunk 3]
```diff
@@ -2922,7 +2922,7 @@
     ],
     "connections": [ { "point": [ -1, 1, 0 ], "terrain": "road", "connection": "local_road", "from": [ 0, 1, 0 ] } ],
     "locations": [ "field", "forest_without_trail" ],
-    "city_distance": [ 5, 40 ],
+    "city_distance": [ 5, 120 ],
```
Comment (Maleclypse) [inline data/json/overmap/overmap_special/specials.json:2925]:
```suggestion
    "city_distance": [ 5, -1 ],
```

File: data/json/overmap/overmap_special/specials.json [hunk 4]
```diff
@@ -2940,7 +2940,7 @@
     ],
     "connections": [ { "point": [ 0, 2, 0 ], "terrain": "road", "connection": "local_road" } ],
     "locations": [ "field" ],
-    "city_distance": [ 5, 60 ],
+    "city_distance": [ 5, 400 ],
```
Comment (Maleclypse) [inline data/json/overmap/overmap_special/specials.json:2943]:
```suggestion
    "city_distance": [ 5, -1 ],
```

File: data/json/overmap/overmap_special/specials.json [hunk 5]
```diff
@@ -4403,7 +4403,7 @@
       { "point": [ 1, -1, 0 ], "terrain": "road", "connection": "local_road" }
     ],
     "locations": [ "land" ],
-    "city_distance": [ 5, 60 ],
+    "city_distance": [ 5, 400 ],
```
Comment (Maleclypse) [inline data/json/overmap/overmap_special/specials.json:4406]:
```suggestion
    "city_distance": [ 5, -1 ],
```

**Comment by LovamKicsiGazsii** [comment]
Thanks for that, @Maleclypse !

**Comment by KeremBabaG** [comment]
> > 400 is too much imo, how about 200?
> 
> I think you are confused. Why would you think that?

yeah I seemingly did. thought that was the minimum

**Pull request merged**